src/NServiceBus.Hosting.Windows/StartableAndStoppable/StartableAndStoppableRunner.cs

```diff
@@ -0,0 +1,103 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Hosting;
+    using Logging;
+
+    class StartableAndStoppableRunner
+    {
+        public StartableAndStoppableRunner(IEnumerable<IWantToRunWhenEndpointStartsAndStops> wantToRunWhenBusStartsAndStops)
+        {
+            this.wantToRunWhenBusStartsAndStops = wantToRunWhenBusStartsAndStops;
+        }
+
+        public Task Start(IMessageSession session)
+        {
+            var startableTasks = new List<Task>();
+            foreach (var startable in wantToRunWhenBusStartsAndStops)
+            {
+                var task = startable.Start(session).ThrowIfNull();
+
+                var startable1 = startable;
+
+                /* 
+                    We can't use the await keyword because of the conditional logging. 
+                    Since we want to start them concurrently and log per instance there is not much else we can do.
+                */
+                task.ContinueWith(t =>
+                {
+                    thingsRanAtStartup.Add(startable1);
+                    Log.DebugFormat("Started {0}.", startable1.GetType().AssemblyQualifiedName);
+                }, TaskContinuationOptions.OnlyOnRanToCompletion | TaskContinuationOptions.ExecuteSynchronously);
+                task.ContinueWith(t =>
+                {
+                    Log.Error($"Startup task {startable1.GetType().AssemblyQualifiedName} failed to complete.", t.Exception);
+                }, TaskContinuationOptions.OnlyOnFaulted | TaskContinuationOptions.ExecuteSynchronously);
+                
+                startableTasks.Add(task);
+            }
+
+            return Task.WhenAll(startableTasks.ToArray());
```

Comment:
@indualagarsamy - @andreasohlund and I discussed this on a zoom call and we agreed that it makes sense to do, but that it's also a different PR since this is about moving the existing behavior from the core to the host. Seems like we forgot the update the discussion here.